Firestore/src/FirestoreSessionHandler.php

```diff
@@ -0,0 +1,404 @@
+        $database = $this->databaseName($this->projectId, $this->database);
+
+        $beginTransaction = $this->connection->beginTransaction([
+            'database' => $database
+        ] + $this->options['begin']);
+
+        $this->transaction = new Transaction(
+            $this->connection,
+            $this->valueMapper,
+            $database,
+            $beginTransaction['transaction']
+        );
+
+        return true;
+    }
+
+    /**
+     * Just return true for this implementation.
+     */
+    public function close()
+    {
+        return true;
+    }
+
+    /**
+     * Read the session data from Cloud Firestore.
+     *
+     * @param string $id Identifier used for the session.
+     * @return string
+     */
+    public function read($id)
+    {
+        try {
+            $docRef = $this->getDocumentReference(
+                $this->connection,
+                $this->valueMapper,
+                $this->projectId,
+                $this->database,
+                $this->docId($id)
+            );
+            $snapshot = $this->transaction->snapshot($docRef);
+            if ($snapshot->exists() && isset($snapshot['data'])) {
+                return $snapshot->get('data');
+            }
+        } catch (ServiceException $e) {
+            trigger_error(
+                sprintf('Firestore lookup failed: %s', $e->getMessage()),
+                E_USER_WARNING
+            );
+        }
+        return '';
+    }
+
+    /**
+     * Write the session data to Cloud Firestore.
+     *
+     * @param string $id Identifier used for the session.
+     * @param string $data The session data to write to the Firestore document.
+     * @return bool
+     */
+    public function write($id, $data)
+    {
+        try {
+            $docRef = $this->getDocumentReference(
+                $this->connection,
+                $this->valueMapper,
+                $this->projectId,
+                $this->database,
+                $this->docId($id)
+            );
+            $this->transaction->set($docRef, [
+                'data' => $data,
+                't' => time()
+            ]);
+            $this->commitTransaction();
+        } catch (ServiceException $e) {
+            trigger_error(
+                sprintf('Firestore upsert failed: %s', $e->getMessage()),
+                E_USER_WARNING
+            );
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Delete the session data from Cloud Firestore.
+     *
+     * @param string $id Identifier used for the session
+     * @return bool
+     */
+    public function destroy($id)
+    {
+        try {
+            $docRef = $this->getDocumentReference(
+                $this->connection,
+                $this->valueMapper,
+                $this->projectId,
+                $this->database,
+                $this->docId($id)
+            );
+            $this->transaction->delete($docRef, $this->options['delete']);
```

Comment:
Since delete is inside the transaction and service calls are only done on `commit`, don't think we need the 2nd parameter here. probably don't need the option at all.